src/test/java/com/cloudant/tests/CloudantClientTestBase.java

```diff
@@ -0,0 +1,82 @@
+package com.cloudant.tests;
+
+import com.cloudant.client.api.CloudantClient;
+
+import org.junit.After;
+import org.junit.Before;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/**
+ * Created by Rhys Short on 21/05/15.
+ */
+public abstract class CloudantClientTestBase {
```

Comment:
Another reason not to have the `@Before` and `@After` annotations on methods in a superclass like this is that it's not possible to override them AFAIK.  I think JUnit will run the superclass version regardless of whether you've overridden them (in a Java sense), and then also run the subclass's `@Before` methods too, so I think the way these tests are currently set up with most of the concrete classes calling `super.setUp()` the version in this class will end up being called twice for each test.